packages/components/src/form-token-field/suggestions-list.js

```diff
@@ -1,158 +1,150 @@
+		setList( ref );
+	};
 
-	handleHover( suggestion ) {
+	const handleHover = ( suggestion ) => {
 		return () => {
-			if ( ! this.scrollingIntoView ) {
-				this.props.onHover( suggestion );
+			if ( ! scrollingIntoView ) {
+				onHover( suggestion );
 			}
 		};
-	}
+	};
 
-	handleClick( suggestion ) {
+	const handleClick = ( suggestion ) => {
 		return () => {
-			this.props.onSelect( suggestion );
+			onSelect( suggestion );
 		};
-	}
-
-	handleMouseDown( e ) {
-		// By preventing default here, we will not lose focus of <input> when clicking a suggestion
-		e.preventDefault();
-	}
+	};
 
-	computeSuggestionMatch( suggestion ) {
-		const match = this.props
-			.displayTransform( this.props.match || '' )
-			.toLocaleLowerCase();
-		if ( match.length === 0 ) {
+	const computeSuggestionMatch = ( suggestion ) => {
+		const matchText = displayTransform( match || '' ).toLocaleLowerCase();
+		if ( matchText.length === 0 ) {
 			return null;
 		}
 
-		suggestion = this.props.displayTransform( suggestion );
-		const indexOfMatch = suggestion.toLocaleLowerCase().indexOf( match );
+		suggestion = displayTransform( suggestion );
+		const indexOfMatch = suggestion
+			.toLocaleLowerCase()
+			.indexOf( matchText );
 
 		return {
 			suggestionBeforeMatch: suggestion.substring( 0, indexOfMatch ),
 			suggestionMatch: suggestion.substring(
 				indexOfMatch,
-				indexOfMatch + match.length
+				indexOfMatch + matchText.length
 			),
 			suggestionAfterMatch: suggestion.substring(
-				indexOfMatch + match.length
+				indexOfMatch + matchText.length
 			),
 		};
-	}
+	};
 
-	render() {
-		// We set `tabIndex` here because otherwise Firefox sets focus on this
-		// div when tabbing off of the input in `TokenField` -- not really sure
-		// why, since usually a div isn't focusable by default
-		// TODO does this still apply now that it's a <ul> and not a <div>?
-		return (
-			<ul
-				ref={ this.bindList }
-				className="components-form-token-field__suggestions-list"
-				id={ `components-form-token-suggestions-${ this.props.instanceId }` }
-				role="listbox"
-			>
-				{ map( this.props.suggestions, ( suggestion, index ) => {
-					const match = this.computeSuggestionMatch( suggestion );
-					const classeName = classnames(
-						'components-form-token-field__suggestion',
-						{
-							'is-selected': index === this.props.selectedIndex,
-						}
-					);
+	// We set `tabIndex` here because otherwise Firefox sets focus on this
+	// div when tabbing off of the input in `TokenField` -- not really sure
+	// why, since usually a div isn't focusable by default
+	// TODO does this still apply now that it's a <ul> and not a <div>?
+	return (
+		<ul
+			ref={ bindList }
+			className="components-form-token-field__suggestions-list"
+			id={ `components-form-token-suggestions-${ instanceId }` }
+			role="listbox"
+		>
+			{ map( suggestions, ( suggestion, index ) => {
+				const matchText = computeSuggestionMatch( suggestion );
+				const classeName = classnames(
```

Comment:
Nit: typo

```suggestion
				const className = classnames(
```

(we should then update any reference to this variable)

Reply:
Apologies for this 😅 

I have tested again my changes. Working as expected. :smile: